compose/utils.py

```diff
@@ -91,3 +93,15 @@ def json_hash(obj):
     h = hashlib.sha256()
     h.update(dump)
     return h.hexdigest()
+
+
+def interpolate(string, mapping):
+    try:
+        return Template(string).substitute(defaultdict(lambda: "", mapping))
+    except ValueError:
```

Comment:
Using `safe_substitute()` would be closer to how `os.path.expandvars()` worked previously (when keys are missing, leave the original value).  

Otherwise this `except` should also include `KeyError`, which is raised when the value isn't in the environment (which I think should be a test case).